### ISSUE TITLE
`regulo/store.py` `load()` silently swaps Covridge/Sparridge gram matrix for identity

### ISSUE THREAD

**Issue by sachncs** [sachncs/regulo#10]
## Problem

`regulo.store.load()` re-derives the geometry-aware penalty's Gram matrix as the identity matrix `np.eye(p)` instead of the original training-data Gram matrix used at save time. A model saved with Covridge or Sparridge therefore loads back with a different penalty than the one used during training, and subsequent fine-tuning, evaluation, or comparison will be wrong.

## Evidence

- `regulo/store.py:162-172` —
  ```python
  def penaltylookup(data: Dict) -> Penalty:
      name = data["penalty"]
      hp = dict(data["penaltyargs"])
      if name in ("covridge", "sparridge"):
          # Re-derive a synthetic gram matrix from shape[0]; the
          # user's actual training data was used to build the original,
          # so the loaded penalty applies only to the first layer.
          p = data["shape"][0]
          hp["gram"] = np.eye(p)
      cls = REGISTRY[name]
      return cls(**hp)
  ```

- The save path serialises the architecture and hyperparameters but not the original Gram matrix. After `load()`, `Covridge.csqrt == np.eye(p)`, so the penalty degenerates into a plain `Ridge` with strength `lambda1 + lambda2` (because the gradient is `2 lambda1 (I @ W) + 2 lambda2 W = 2 (lambda1 + lambda2) W`).
- `tests/test_store.py:29-59` (`test_saveloadroundtrip`) only round-trips weights / biases / Adam state. There is no test that verifies the loaded penalty's value matches the saved penalty's value.

## Root Cause

`save()` does not persist the Gram matrix used at construction time. `load()` substitutes an identity matrix, which does not match the training data and changes the penalty's mathematical behaviour.

## Impact

- Anyone who serialises a model trained with `Covridge` or `Sparridge` and reloads it later (e.g. for inference, ensembling, or resuming training) will use a penalty that is no longer the one that was trained. The model's predictions will silently drift.
- The library advertises `docs/limits.md:33-35` ("No model versioning beyond `__version__`. `save` / `load` rejects mismatched major versions") — silent penalty substitution is a worse failure mode than a version error.

## Recommendation

Persist the Gram matrix in `save()` (e.g. `gram.npy` alongside `weights.npz`) and load it back in `load()` when the penalty is Covridge or Sparridge. Reject the load with a clear `ValueError` if the shape does not match `shape[0]`.

For users who genuinely want identity-Gram load, expose an opt-in `load(path, restore_gram=False)` flag.

## Acceptance Criteria

- [ ] After `save(runner_with_covridge, path); runner2 = load(path)`, `np.allclose(runner2.penalty.csqrt, original_penalty.csqrt)`.
- [ ] After `save(runner_with_sparridge, path); runner2 = load(path)`, same equality holds for Sparridge.
- [ ] `tests/test_store.py` includes a regression test for Covridge / Sparridge round-trip equality.
- [ ] If persistence of the Gram matrix is rejected for size reasons, `load()` raises a clear `ValueError` instead of silently substituting identity.

## Verification

- [ ] Round-trip test passes for all six penalties.
- [ ] Manual: train a Covridge model, save, load, and confirm the loaded penalty evaluates the same value on a fixed weight matrix.

## Risk

Breaking — any user who has previously relied on identity-Gram reload will need to migrate. Mitigation: introduce the new behaviour in `v0.3.0` (minor bump) and document in `CHANGELOG.md`.

**Comment by sachncs** [comment]
Resolved in PR #29.
